File: src/test/resources/integration_tests/javascript/passIntegrationTest.js

var container = require("vertx/container")
var vertx = require("vertx");
var vertxTests = require("vertx_tests");
var vassert = require("vertx_assert");

function testAssertOK() {
  vassert.assertEquals("foo", "foo")
  vassert.testComplete()
}

function testAssertOKFromHandler() {
  vertx.runOnContext(function() {
    vassert.assertEquals("foo", "foo")
    vassert.testComplete()
  });
}

function testAssertOKFromVerticle() {
  container.deployVerticle("test_verticle.js", {type:"assert_ok"})
}

vertxTests.startTests(this);

File: src/test/resources/integration_tests/python/pass_integration_test.py

from org.vertx.testtools import VertxAssert
import vertx_tests
from core.event_bus import EventBus
import vertx

def test_assert_ok():
    VertxAssert.assertEquals('foo', 'foo')
    VertxAssert.testComplete()

def test_assert_ok_from_handler():
    @vertx.run_on_context
    def handler():
        VertxAssert.assertEquals("foo", "foo")
        VertxAssert.testComplete()

def test_assert_ok_from_verticle():
    conf = {'type' : 'assert_ok'}
    vertx.deploy_verticle('test_verticle.py', conf)


vertx_tests.start_tests(locals())



